### PR TITLE
kick-off compatibility matrix

### PR DESCRIPTION
**Pull request by amanusk**
### Description of the Changes

Please add a detailed description of the change, whether it's an enhancement or a bugfix.
If the PR is related to an open issue please link to it.

### PR Preview URL

After you push a commit to this PR, a preview is built and a URL to the root of the preview appears in the comment feed.

Paste here the specific URL(s) of the content that this PR addresses.

### Check List

- [X] Changes have been done against main branch, and PR does not conflict
- [X] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1347)

### CODE REVIEW

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+|===
+
+== Cairo
+
+|===
+| | 2.8.0 | 2.7.0 | 2.6.0
+| starknet.js | 6.11.0 |  | 
+| starknet.py | 0.21.0 |  | 
+| starkli | 0.3.4 | 0.3.4 | 0.3.3
+| Scarb | 2.8.0 | 2.7.1 |
```
Comment (maciektr) [inline components/Starknet/modules/tools/pages/compatibility.adoc:56]:
```suggestion
| Scarb | 2.8.1 | 2.7.1 | 2.6.5
```

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+|===
+| | RPC 0.7 | RPC 0.6
+| starknet.js | ≥ 6.11.0 | 5.14.0
+| starknet.py | ≥ 0.21.0 | 0.20.0
+| starknet-rs | ≥ 0.11.0 | 0.10.0
+| starknet.go | ≥ 0.7.0 | 0.6.1
+| starkli | ≥ 0.3.0 | 0.2.0
+| Pathfinder | ≥ 0.11.0 | 0.10.2
+| Juno | ≥ 0.11.0 | 0.8.0
+| Starknet Foundry | ≥ 0.28.0 | 0.19.0
```
Comment (Arcticae) [inline components/Starknet/modules/tools/pages/compatibility.adoc:45]:
```suggestion
| Starknet Foundry | ≥ 0.27.0 | 0.20.0
```

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+== Cairo
+
+|===
+| | 2.8.0 | 2.7.0 | 2.6.0
+| starknet.js | 6.11.0 |  | 
+| starknet.py | 0.21.0 |  | 
+| starkli | 0.3.4 | 0.3.4 | 0.3.3
+| Scarb | 2.8.0 | 2.7.1 | 
+| Starknet Foundry | 0.29.0 | 0.28.0 | 0.27.0
+| starknet-devnet |  |  | 0.1.4
```
Comment (FabijanC) [inline components/Starknet/modules/tools/pages/compatibility.adoc:58]: Here it's `starknet-devnet` and in the table above it's `starknet-devnet-rs`. Also, `starknet-devnet-rs` 0.1.4 does not exist. Due to the way [Universal Sierra Compiler](https://github.com/software-mansion/universal-sierra-compiler) works, all past versions of Sierra should be supported.

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+== Cairo
+
+|===
+| | 2.8.0 | 2.7.0 | 2.6.0
+| starknet.js | 6.11.0 |  | 
+| starknet.py | 0.21.0 |  | 
+| starkli | 0.3.4 | 0.3.4 | 0.3.3
+| Scarb | 2.8.0 | 2.7.1 | 
+| Starknet Foundry | 0.29.0 | 0.28.0 | 0.27.0
```
Comment (Arcticae) [inline components/Starknet/modules/tools/pages/compatibility.adoc:57]: There's no strict relation here but it's _supposed to_ be compatible like this
```suggestion
| Starknet Foundry | >=0.28.0 | >=0.28.0 | 0.27.0
```

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+| | RPC 0.7 | RPC 0.6
+| starknet.js | ≥ 6.11.0 | 5.14.0
+| starknet.py | ≥ 0.21.0 | 0.20.0
+| starknet-rs | ≥ 0.11.0 | 0.10.0
+| starknet.go | ≥ 0.7.0 | 0.6.1
+| starkli | ≥ 0.3.0 | 0.2.0
+| Pathfinder | ≥ 0.11.0 | 0.10.2
+| Juno | ≥ 0.11.0 | 0.8.0
+| Starknet Foundry | ≥ 0.28.0 | 0.19.0
+| starknet-devnet-rs | ≥ 0.1.4 | 0.1.0
```
Comment (FabijanC) [inline components/Starknet/modules/tools/pages/compatibility.adoc:46]: The last Devnet with RPC 0.6 support is [v0.0.2](https://github.com/0xSpaceShard/starknet-devnet-rs/releases/tag/v0.0.2). The first with RPC 0.7 support is [v0.0.4](https://github.com/0xSpaceShard/starknet-devnet-rs/releases/tag/v0.0.4). Between them, release candidates for 0.7 were supported.

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+|===
+
+== Cairo
+
+|===
+| | 2.8.0 | 2.7.0 | 2.6.0
+| starknet.js | 6.11.0 |  | 
+| starknet.py | 0.21.0 |  |
```
Comment (franciszekjob) [inline components/Starknet/modules/tools/pages/compatibility.adoc:54]:
```suggestion
| starknet.py | ≥ 0.24.1 | ≥ 0.24.1 | 0.19.0
| starknet-jvm | ≥ 0.11.1 | ≥ 0.11.1 | ≥ 0.11.1
| starknet.swift | ≥ 0.9.0 | ≥ 0.9.0 | ≥ 0.9.0
```

File: components/Starknet/modules/tools/pages/compatibility.adoc [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+|===
+
+== Starknet API
+
+|===
+| | RPC 0.7 | RPC 0.6
+| starknet.js | ≥ 6.11.0 | 5.14.0
+| starknet.py | ≥ 0.21.0 | 0.20.0
+| starknet-rs | ≥ 0.11.0 | 0.10.0
+| starknet.go | ≥ 0.7.0 | 0.6.1
```
Comment (franciszekjob) [inline components/Starknet/modules/tools/pages/compatibility.adoc:41]:
```suggestion
| starknet.go | ≥ 0.7.0 | 0.6.1
| starknet-jvm | ≥ 0.11.0 | 0.10.0
| starknet.swift | ≥ 0.9.0 | 0.8.0
```

File: components/Starknet/modules/tools/pages/compatibility.adoc (hunk 3 above)
Comment (amanusk) [reply components/Starknet/modules/tools/pages/compatibility.adoc:62]: Does it make more sense to move devnet to the upper table, with other full nodes?

**Comment by LandauRaz** [comment]
Hey @amanusk, let's talk about this PR privately (closing it in the meantime).

**Comment by LandauRaz** [comment]
Addressed in #1417.

**Pull request closed without merging**